terragrunt/aws/cdn/cloudfront.tf

```diff
@@ -46,3 +51,38 @@ resource "aws_cloudfront_distribution" "cdn" {
     Terraform  = true
   }
 }
+
+# Bucket to store cloudfront logscheck "name" {
+module "cloudfront_logs" {
+  source            = "github.com/cds-snc/terraform-modules//S3?ref=v9.4.4"
+  bucket_name       = "${var.product_name}-${var.env}-cdn-logs"
+  billing_tag_value = var.billing_code
+
+}
+
+# Bucket policy to allow CloudFront to write logs
+resource "aws_s3_bucket_policy" "cloudfront_logs_policy" {
+  bucket = module.cloudfront_logs.s3_bucket_id
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Principal = {
+          Service = "cloudfront.amazonaws.com"
+        },
+        Action   = "s3:PutObject",
+        Resource = "arn:aws:s3:::${module.cloudfront_logs.s3_bucket_id}/*",
+        Condition = {
+          StringEquals = {
+            "AWS:SourceArn" = aws_cloudfront_distribution.cdn.arn
+          }
+        }
+      }
+    ]
+  })
+
+  # Add explicit dependency to ensure CloudFront exists before updating the policy
+  depends_on = [aws_cloudfront_distribution.cdn]
```

Comment:
You should be able to get rid of this `depends_on` because of the implicit dependency above where the ARN is referenced for the `AWS:SourceArn` condition:
```suggestion
```

Reply:
Thanks for the suggestion - I have committed your change.